src/protocols/rdp/channels/rail.c

```diff
@@ -213,9 +208,63 @@ static UINT guac_rdp_rail_handshake(RailClientContext* rail,
  */
 static UINT guac_rdp_rail_handshake_ex(RailClientContext* rail,
         RAIL_CONST RAIL_HANDSHAKE_EX_ORDER* handshake_ex) {
+    guac_client* client = (guac_client*) rail->custom;
+    guac_client_log(client, GUAC_LOG_TRACE, "RAIL handshake ex callback.");
     return guac_rdp_rail_complete_handshake(rail);
 }
 
+/**
+ * A callback function that is executed when an update for a RAIL window is
+ * received from the RDP server.
+ *
+ * @param context
+ *     A pointer to the rdpContext structure used by FreeRDP to handle the
+ *     window update.
+ *
+ * @param orderInfo
+ *     A pointer to the data structure that contains information about what
+ *     window was updated what updates were performed.
+ *
+ * @param windowState
+ *     A pointer to the data structure that contains details of the updates
+ *     to the window, as indicated by flags in the orderInfo field.
+ *
+ * @return
+ *     TRUE if the client-side processing of the updates as successful; otherwise
```

Comment:
Looks to me like it always returns true,

Reply:
Clarified via rebase.